setuptools/tests/config/test_setupcfg.py

```diff
@@ -716,6 +716,41 @@ def test_extras_require(self, tmpdir):
             }
             assert dist.metadata.provides_extras == set(['pdf', 'rest'])
 
+    @pytest.mark.parametrize(
+        "config",
+        [
+            "[options.extras_require]\nfoo = bar;python_version<'3'",
+            "[options.extras_require]\nfoo = bar;os_name=='linux'"
+        ]
+    )
+    def test_extras_warns_about_potential_misconfig(self, config, tmpdir):
+        fake_env(tmpdir, config)
+
+        match = ("One of the parsed requirements in extras_require "
+                 "section looks like a valid environment marker.*")
+        with pytest.warns(UserWarning, match=match):
+            with get_dist(tmpdir) as dist:
+                # We are testing the warning,
+                # the result should be the same
+                assert len(dist.extras_require) == 1
+                assert len(dist.extras_require["foo"]) == 2
+
+    @pytest.mark.parametrize(
+        "config",
+        [
+            "[options.extras_require]\nfoo =\n    bar;python_version<'3'",
+            "[options.extras_require]\nfoo = bar;baz\nboo = xxx;yyy"
```

Comment:
Maybe we could also add here another valid example:

```ini
[options.extras_require]
foo =
    bar
    python_version<'3'
```
(https://pypi.org/project/python_version/)